fix(curriculum): add test to ensure typeAttr matches srcAttr media type

l3onhard opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitHub Codespaces.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #62503

<!-- Feel free to add any additional description of changes below this line -->

For the `<source>` element inside the `<video>` element: I added a test to assert equality of the `type` attribute and the media type of the `src` attribute.

l3onhard committed:
fix(curriculum): add test to ensure source element type matches src media type

l3onhard commented:
@ilenia-magoni, you are right: Matching the slash is unnecessary. The regex should be: `/[^\/\.]+$/g`.





> Am 04.10.2025 um 18:24 schrieb Ilenia ***@***.***>:
> 
> 
> @ilenia-magoni commented on this pull request.
> 
> In curriculum/challenges/english/blocks/lab-html-audio-and-video-player/68de40f53fcaf106f51cc20f.md <https://github.com/freeCodeCamp/freeCodeCamp/pull/62512#discussion_r2404077070>:
> 
> > @@ -112,6 +112,17 @@ const typeAttr = source.getAttribute('type');
>  assert.isNotEmpty(typeAttr);
>  ```
>  
> +The `type` attribute should match the media type of the file extension of the `src` attribute.
> +
> +```js
> +const section = document.querySelector('section');
> +const source = section?.querySelector('h2 + video > source');
> +const typeAttr = source.getAttribute('type');
> +const srcAttr = source.getAttribute('src');
> +const srcMediaType = 'video/' + srcAttr.match(/[^\/\.]+$/g)[0];
> why are you matching a slash before the dot here?
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/freeCodeCamp/freeCodeCamp/pull/62512#pullrequestreview-3301994851>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ACUFEM222BLP4FLW5P6NMG33V7YFHAVCNFSM6AAAAACIJDM6K2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTGMBRHE4TIOBVGE>.
> You are receiving this because you authored the thread.
>

ilenia-magoni commented:
I am marking this as waiting update, until you make that change

l3onhard committed:
fix: remove unnecessary part of the regex

ilenia-magoni commented:
unfortunately I have to remove my approval, the `type` attribute can't be written with just the extension of the file, see https://developer.mozilla.org/en-US/docs/Web/Media/Guides/Formats/Containers#common_container_formats

l3onhard commented:
Thank you for catching that mistake. I didn't know. 

Now that I am researching how to map file extensions to media types, I would suggest that we also change the following test (starting line 97): 

> The `source` element should have a `src` attribute pointing to a video file.
> 
> ```js
> const section = document.querySelector('section');
> const source = section?.querySelector('h2 + video > source');
> const srcAttr = source.getAttribute('src');
> assert.match(srcAttr, /\.(mp4|webm|ogg|avi|mov)$/i);
> ```

The file extension `.ogg` (among others) is for the media type `audio/ogg`. The file extension for `video/ogg` is `.ogv`. Here are my sources: 

- https://www.iana.org/assignments/media-types/video/ogg 
- https://www.iana.org/assignments/media-types/audio/ogg.
- https://wiki.xiph.org/index.php/MIME_Types_and_File_Extensions

Therefore, I would suggest that we change `ogg` to `ogv` in the test above.

l3onhard commented:
Concerning the file extension `.avi`: 

According to https://www.jmcgowan.com/aviweb.html#MIME these media types are used for it: 
- video/avi
- video/msvideo
- video/x-msvideo

According to https://www.rfc-editor.org/rfc/rfc2361: "Internet applications may reference specific codecs within the WAVE and AVI registries as follows: video/vnd.avi [...]"

Therefore, I would suggest that we check for all four media types, when the file extension `.avi` is used.

l3onhard commented:
We can ignore all other file extensions (e.g. `.opus` or `.m4v`),  because they are excluded by the test I already mentioned above:

> The `source` element should have a `src` attribute pointing to a video file.
> 
> ```js
> const section = document.querySelector('section');
> const source = section?.querySelector('h2 + video > source');
> const srcAttr = source.getAttribute('src');
> assert.match(srcAttr, /\.(mp4|webm|ogg|avi|mov)$/i);
> ```

l3onhard committed:
fix: add file extension to media types mapping

See my comments in the PR #62512: 
https://github.com/freeCodeCamp/freeCodeCamp/pull/62512#issuecomment-3369293865
https://github.com/freeCodeCamp/freeCodeCamp/pull/62512#issuecomment-3369296494
https://github.com/freeCodeCamp/freeCodeCamp/pull/62512#issuecomment-3369304939

majestic-owl448 committed:
Update curriculum/challenges/english/blocks/lab-html-audio-and-video-player/68de40f53fcaf106f51cc20f.md

majestic-owl448 committed:
Update curriculum/challenges/english/blocks/lab-html-audio-and-video-player/68de40f53fcaf106f51cc20f.md

l3onhard committed:
fix(curriculum): use standard spacing

Co-authored-by: Huyen Nguyen <25715018+huyenltnguyen@users.noreply.github.com>

l3onhard commented on curriculum/challenges/english/blocks/lab-html-audio-and-video-player/68de40f53fcaf106f51cc20f.md:
```diff
@@ -112,6 +112,25 @@ const typeAttr = source.getAttribute('type');
+  mov:    ['video/quicktime']
+};
+const srcMediaTypes = fileExtToMediaTypes[srcFileExt];
+assert.oneOf(typeAttr, fileExtToMediaTypes);
```
Why can't I commit this suggestion? I would like to.
